lib/dest/write-contents/write-buffer.js

```diff
@@ -1,14 +1,30 @@
 'use strict';
 
+var iconv = require('iconv-lite');
+
 var fo = require('../../file-operations');
+var getCodec = require('../../codecs');
 
 function writeBuffer(file, optResolver, onWritten) {
+  var encoding = optResolver.resolve('encoding', file);
+  var codec = getCodec(encoding);
```

Comment:
Sorry, I misunderstood, I now realize you were saying `getCodec` could be used as the coercer function passed to value-or-function via resolve-options.

That makes a lot of sense, yeah. I'll try if I can get that to work, nice exercise also to see about error handling in resolve-options, which I wanted to do anyway.